README.md

```diff
@@ -362,29 +368,61 @@ Benchee.run %{
 
-Is executed after a scenario completed and is passed whatever the return value of the last `before_scenario` was, if there was none it is passed the specified `input`. Otherwise it works much like [`after_each`](#after_each).
+After hooks do not form a chain, and their return values are discarded. An `after_each` hook receives the return value of the benchmarking function as its argument. An `after_scenario` function receives the return value of its `before_scenario` counterpart (or the scenario's input if there is no `before_scenario` hook).
+
+If you haven't defined any inputs, the hook chain is started with the special input argument returned by `Benchee.Benchmark.no_input()`.
+
+#### Hook configuration: global versus local
+
+Hooks can be defined either _globally_ as part of the configuration or _locally_ for a specific benchmarking function. Global hooks will be executed for every scenario in the suite. Local hooks will only be executed for scenarios involving that benchmarking function.
+
+**Global hooks**
+
+Global hooks are specified as part of the general benchee configuration:
 
 ```elixir
 Benchee.run %{
-  "bar" => fn -> bar() end
-}, after_scenario: fn(_input) -> bust_my_cache() end
+  "foo" => fn(input) -> ... end,
+  "bar" => fn(input) -> ... end
+}, inputs: %{
+     "input 1" => 1,
+     "input 2" => 2,
+  },
+  before_scenario: fn(input) -> ... end
 ```
 
-When might this be useful?
-* Busting caches after a scenario completed
-* Deleting all the records in the database that the scenario just created
-* Terminating whatever was setup by `before_scenario`
+Here the `before_scenario` function will be executed for all 4 scenarios present in this benchmarking suite.
+
+**Local hooks**
+
+Local hooks are defined alongside the benchmarking function. To define a local hook, pass a tuple in the initial map, instead of just a single function. The benchmarking function comes first, followed by a keyword list specifying the hooks to run:
+
+```elixir
+Benchee.run %{
+  "foo" => {fn(input) -> ... end, before_scenario: fn(input) -> ... end},
+  "bar" => fn(input) -> ... end
+}, inputs: %{
+  "input 1" => 1,
+  "input 2" => 2
+}
+```
+
+Here `before_scenario` is only run for the 2 scenarios associated with `"foo"`, i.e. foo with input 1 and foo with input 2. It is _not_ run for any `"bar"` benchmarks.
+
 
 #### When does a hook happen? (Complete Example)
 
-Yes the whole hooks system is quite a lot to take in. Here is an overview when what hook is executed with what input depending on its previous hooks. In general, if some hook isn't specified the value of the _previous_ hook is passed on.
+Yes the whole hooks system is quite a lot to take in. Here is an overview showing the order of hook execution, along with the argument each hook receives (see [hook arguments and return values](#hook-arguments-and-return-values)).
```

Comment:
For the whole is "global or local invoked first" we also might mention the guiding principle, which is: "hooks defined locally execute closer to the benchmarking function" (which is why it swtiches between before_ and after_)